src/components/scene/pool.js

```diff
@@ -0,0 +1,65 @@
+var debug = require('../../utils/debug');
+var register = require('../../core/component').registerComponent;
+
+var warn = debug('components:pool:warn');
+
+module.exports.Component = register('pool', {
+  schema: {
+    mixin: {default: ''},
+    size: {default: 0}
+  },
```

Comment:
Add another parameter to force the `requestEntity` to return a new entity even if there's no available entity on the pool, it will create a new one and pushed to the pool and return it